File: csharp/GameOfLife/Life.cs
using System;
using System.Collections.Generic;

namespace GameOfLife
{
    public class Life
    {
        private readonly HashSet<Cell> liveCells;

        public Life(HashSet<Cell> initialLiveCells)
        {
            this.liveCells = initialLiveCells;
        }

        public HashSet<Cell> GetLiveCells()
        {
            return this.liveCells;
        }

        public static bool CellShouldSurvive(int numNeighbours) 
        {
            throw new NotImplementedException();
        }
    }
}

File: deprecated/javascript-jasmine/script/life.js
/**
 * Life in Conway's Game of Life.
 */
 var Life = function() {
	
	this.init = function(initialLiveCells /* HashSet<Cell> */) {
		this._liveCells = initialLiveCells;
	}

	this.getLiveCells = function() {
		return this._liveCells;
	}
	
	this.cellShouldSurvive = function(numNeighbours /* int */) {

		throw "Unsupported Operation Exception";
	}
	
	this.init.apply(this, arguments);
};
